From 1e044812598485e78f477e315f97175f7d8a0b6b Mon Sep 17 00:00:00 2001
From: Sridharan S N <quic_sridsn@quicinc.com>
Date: Fri, 10 Nov 2023 15:07:49 +0530
Subject: [PATCH] coresight: forward declaration for coresight_abort function

coresight_abort() function is used by the drivers like CNSS2, which
does the forward declaration also. Move the forward declaration
to the coresight header file, so that drivers can just include the
coresight header file. And also add fix to compile with CORESIGHT
disabled.

Change-Id: Ibca9e5392a8779ee958440ebe48b4604cc2d2e20
Signed-off-by: Sridharan S N <quic_sridsn@quicinc.com>
---
 include/linux/coresight.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/coresight.h b/include/linux/coresight.h
index 0ad400464115..37592b4f8900 100644
--- a/include/linux/coresight.h
+++ b/include/linux/coresight.h
@@ -514,6 +514,7 @@ extern char *coresight_alloc_device_name(struct coresight_dev_list *devs,
 					 struct device *dev);
 
 extern bool coresight_loses_context_with_cpu(struct device *dev);
+extern void coresight_abort(void);
 
 u32 coresight_relaxed_read32(struct coresight_device *csdev, u32 offset);
 u32 coresight_read32(struct coresight_device *csdev, u32 offset);
@@ -552,6 +553,7 @@ static inline int coresight_claim_device(struct coresight_device *csdev)
 
 static inline void coresight_disclaim_device(struct coresight_device *csdev) {}
 static inline void coresight_disclaim_device_unlocked(struct coresight_device *csdev) {}
+static inline void coresight_abort(void) {}
 
 static inline bool coresight_loses_context_with_cpu(struct device *dev)
 {
-- 
2.34.1

